crates/swc/tests/tsc-references/ClassAndModuleThatMergeWithStaticFunctionAndExportedFunctionThatShareAName.2.minified.js

```diff
@@ -12,12 +12,9 @@ var A, Point = function() {
         };
     }, Point;
 }();
-!function(Point) {
-    var Origin = function() {
-        return null;
-    };
-    Point.Origin = Origin;
-}(Point || (Point = {})), function(A) {
+(Point || (Point = {})).Origin = function() {
```

Comment:
Not sure if we want to maintain the function's `name`

Reply:
It depends on the configuration